android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java

```diff
@@ -292,13 +292,35 @@ public View getView(int position, View convertView, ViewGroup parent) {
 			if (content != null) {
 				TiBaseListViewItem itemContent = (TiBaseListViewItem) content.findViewById(listContentId);
 				section.populateViews(data, itemContent, template, sectionItemIndex, sectionIndex, content);
+				//Manually add drawable for ripple touch feedback
+				if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+					try {
+						TypedValue typedValue = new TypedValue();
+						context.getTheme().resolveAttribute(android.R.attr.selectableItemBackground, typedValue, true);
+						itemContent.setClickable(true);
+						itemContent.setForeground(context.getResources().getDrawable(typedValue.resourceId, context.getTheme()));
+					} catch (Exception ex) {
+						ex.printStackTrace();
+					}
+				}
 			} else {
 				content = inflater.inflate(listItemId, null);
 				TiBaseListViewItem itemContent = (TiBaseListViewItem) content.findViewById(listContentId);
 				LayoutParams params = new LayoutParams();
 				params.autoFillsWidth = true;
 				itemContent.setLayoutParams(params);
 				section.generateCellContent(sectionIndex, data, template, itemContent, sectionItemIndex, content);
+				//Manually add drawable for ripple touch feedback
+				if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+					try {
+						TypedValue typedValue = new TypedValue();
+						context.getTheme().resolveAttribute(android.R.attr.selectableItemBackground, typedValue, true);
+						itemContent.setClickable(true);
+						itemContent.setForeground(context.getResources().getDrawable(typedValue.resourceId, context.getTheme()));
+					} catch (Exception ex) {
+						ex.printStackTrace();
+					}
+				}
```

Comment:
Ideally, we shouldn't duplicate this code (I know I was guilty of this in my original code experiment), but I'm fine with it for now. We know we need to find a better way to make this work on older Android OS versions anyways.

Thanks for digging into this @ypbnv! :)